loopy/target/c/codegen/expression.py

```diff
@@ -544,6 +548,19 @@ def map_group_hw_index(self, expr, type_context):
     def map_local_hw_index(self, expr, type_context):
         raise LoopyError("plain C does not have local hw axes")
 
+    def map_nan(self, expr, type_context):
+        from loopy.types import NumpyType
+        if expr.data_type in [None, np.float64]:
+            return p.Variable("NAN")
```

Comment:
This should consider the surrounding `type_context`.

Reply:
Not quite sure of the implementation here. Gave it a shot in https://github.com/inducer/loopy/compare/c47233ea3eb391b94e2b7ed62a48bae893288f76..b08b489e46d1694d1cd3f050260b5760a6ef07e1. Any suggestions?

Reply:
Not quite. The intention of `type_context` is to avoid (likely unintentional) generation of double precision code for otherwise untyped floating point constructs. In particular, if the literal provides a type, that *must* be obeyed. (Right now, it seems like that's ignored if `type_context == "d"`.

Reply:
Yep, fixed. Thanks! (also added a note in the docs about the purpose of `type_context`)